pkg/jwt/jwtToken.go

```diff
@@ -0,0 +1,46 @@
+package jwt
+
+import (
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"time"
+
+	"github.com/golang-jwt/jwt/v5"
+)
+
+type JwtIssuer struct {
+	issuer string
+	secret string
+	ttl    time.Time
+}
+
+func NewJwtIssuer(issuerId string,secretKey string,ttl time.Time) (*JwtIssuer) {
+	return &JwtIssuer{
+		issuer: issuerId,
+		secret:secretKey,
+		ttl: ttl,
+	}
+}
+func (j *JwtIssuer) GeneratedJwtToken(extras map[string]string) (string, error) {
+	pemData, err := base64.RawStdEncoding.DecodeString(j.secret)
+	if len(pemData) == 0 {
+		return "", fmt.Errorf("failed to decode pem: %w", err)
+	}
+	block, _ := pem.Decode(pemData)
+	if block == nil {
+		return "", fmt.Errorf("failed to decode pem")
+	}
+	//create token
+	token := jwt.NewWithClaims(jwt.SigningMethodRS256, jwt.RegisteredClaims{
+		IssuedAt:  jwt.NewNumericDate(time.Now()),
```

Comment:
since we are gonna use `time.Now()` in ExpiresAt as well we need to use in both the same value. so I would assign `time.Now()` to a new variable and use here

Reply:
I nice catch and implemented